packages/jellyfish-api-core/__tests__/category/masternode/setGov.locks.token.test.ts

```diff
@@ -0,0 +1,617 @@
+    await testing.rpc.masternode.setGov({ ATTRIBUTES: { [`v0/locks/token/${tslaId}`]: 'false' } })
+    await testing.generate(1)
+
+    {
+      const attributes = await testing.rpc.masternode.getGov('ATTRIBUTES')
+      expect(
+        attributes.ATTRIBUTES[`v0/locks/token/${tslaId}`] === 'true' ||
+        attributes.ATTRIBUTES[`v0/locks/token/${tslaId}`] === 'false'
+      ).toStrictEqual(true)
+    }
+
+    // Unlock and lock loan tokens in the same block
+    await testing.rpc.masternode.setGov({ ATTRIBUTES: { [`v0/locks/token/${tslaId}`]: 'false' } })
+    await testing.rpc.masternode.setGov({ ATTRIBUTES: { [`v0/locks/token/${tslaId}`]: 'true' } })
+    await testing.generate(1)
+
+    {
+      const attributes = await testing.rpc.masternode.getGov('ATTRIBUTES')
+      expect(
+        attributes.ATTRIBUTES[`v0/locks/token/${tslaId}`] === 'true' ||
+        attributes.ATTRIBUTES[`v0/locks/token/${tslaId}`] === 'false'
+      ).toStrictEqual(true)
+    }
+
+    // 2 lock loan tokens in the same block
+    await testing.rpc.masternode.setGov({ ATTRIBUTES: { [`v0/locks/token/${tslaId}`]: 'true' } })
+    await testing.rpc.masternode.setGov({ ATTRIBUTES: { [`v0/locks/token/${tslaId}`]: 'true' } })
+    await testing.generate(1)
+
+    {
+      const attributes = await testing.rpc.masternode.getGov('ATTRIBUTES')
+      expect(attributes.ATTRIBUTES[`v0/locks/token/${tslaId}`]).toStrictEqual('true')
+    }
+
+    // 2 unlock loan tokens in the same block
+    await testing.rpc.masternode.setGov({ ATTRIBUTES: { [`v0/locks/token/${tslaId}`]: 'false' } })
+    await testing.rpc.masternode.setGov({ ATTRIBUTES: { [`v0/locks/token/${tslaId}`]: 'false' } })
+    await testing.generate(1)
+
+    {
+      const attributes = await testing.rpc.masternode.getGov('ATTRIBUTES')
+      expect(attributes.ATTRIBUTES[`v0/locks/token/${tslaId}`]).toStrictEqual('false')
+    }
+
+    // Lock invalid loan token string
+    await testing.rpc.masternode.setGov({ ATTRIBUTES: { 'v0/locks/token/abc': 'true' } })
+    await testing.generate(1)
+
+    {
+      const attributes = await testing.rpc.masternode.getGov('ATTRIBUTES')
+      expect(attributes.ATTRIBUTES['v0/locks/token/abc']).toBeUndefined()
+    }
+
+    // Unlock invalid loan token string
+    await testing.rpc.masternode.setGov({ ATTRIBUTES: { 'v0/locks/token/abc': 'false' } })
+    await testing.generate(1)
+
+    {
+      const attributes = await testing.rpc.masternode.getGov('ATTRIBUTES')
+      expect(attributes.ATTRIBUTES['v0/locks/token/abc']).toBeUndefined()
+    }
+  })
+
+  it('should update loan token if loan token is unlocked', async () => {
+    // Initially keep the token in locked state
+    await testing.rpc.masternode.setGov({ ATTRIBUTES: { [`v0/locks/token/${tslaId}`]: 'true' } })
+    await testing.generate(1)
+
+    // Unlock loan token
+    await testing.rpc.masternode.setGov({ ATTRIBUTES: { [`v0/locks/token/${tslaId}`]: 'false' } })
```

Comment:
this test and the negative test could have been written in a single test. But it's fine

Reply:
But you did not do this in your loan PR last year. All of us follow this way for a very long time.

Reply:
token split tests are more than just RPC testsing. In a RPC tests, you only target the functionality of the RPC.
But In a token split test, you should check 
1. various actions on entities like vault, acution is blocked after the split, within the locked period.
2. Then you update the oracle prices, token unlock -> system should be back to its normal functionalities.

the structure is there to help the quality of the code. if you only achieve the structure but not the quality, it's incorrect.

Reply:
I mean this => this test and the negative test could have been written in a single test.

Reply:
Your other comment maybe correct

Reply:
Why can't we achieve structure and quality together?

Reply:
Have you added the code for this?

1) various actions on entities like vault, acution is blocked after the split, within the locked period.

2) Then you update the oracle prices, token unlock -> system should be back to its normal functionalities.

Reply:
Usually, we do naming on a file for RPC like getPoolpair.test.ts is for getPoolpair RPC. When you mentioned it should be token split, yes I totally agree with it. However, `v0/oracles/splits/` it is written oracles splits here. So, something wrong with the naming in C++ side.  Like what I wrote yesterday

Should we change v0/oracles/splits/ to v0/tokens/splits/ , I feel we should. @surangap Can you advise C++ side on this?

Reply:
As you have changed the file naming here, then you should advise C++ side. Or maybe there is some reason why they name it oracles split?